### PR TITLE
Updated Validation

### PR DESCRIPTION
**Pull request by nxtloveev3**
Updated validation steps for solute charge and spin multiplicity.

### CODE REVIEW

File: autosolvate/autosolvate.py [hunk 1]
```diff
@@ -755,17 +745,31 @@ def startboxgen(argumentList):
...
+                slu_spinmult = unpaired_electrons + 1
+                if slu_spinmult > 1: charge_method = 'resp'
+                cube_size = solS.get_box_length()
         elif currentArgument in ("-v", "--validation"):
-            sol=PubChemAPI(currentArgument)
-            info=sol.get_info()
-            solS=Solute(info[0], info[1], info[2], info[3])
-            mult_suggest = solS.get_spin_multiplicity() % 2
+            solS = Solute("", "", slu_netcharge, solutexyz)
+            mult_suggest = (abs(slu_netcharge) + 1) % 2
```
Comment (ffangliu) [inline autosolvate/autosolvate.py:768]: This line doesn't make sense... The slu_netcharge is not the number of unpaired electrons. The number of unpaired electrons should be calculated as (total_electron_assuming_neural + netcharge)%2. 
Say, CH3, assuming neural, has 6+3=9 electrons. If the user provides a net charge of -1, then the actual total number of electrons is 9+1 = 10. Then, the # unpaired electron is 10%2 = 0. The spin multiplicity is 2*0+1 = 1.

You can verify for the CH3 (net charge -1), the implementation above is not correct and have multi_suggest = 0.

Pybel can count the number of electrons based on the input xyz file.

**Review by ffangliu — CHANGES_REQUESTED** [review]

**Comment by ffangliu** [comment]
@nxtloveev3 
There are still bugs in the pull request.

**Comment by nxtloveev3** [comment]
The new version is updated and tested using CH3 radical, CH3-, Naphthalene_neutral, and Naphthalene_radical. All of them passed the tests.

File: autosolvate/autosolvate.py [hunk 2]
```diff
@@ -653,8 +653,8 @@ def startboxgen(argumentList):
         Generates the structure files and save as ```.pdb```. Generates the MD parameter-topology and coordinates files and saves as ```.prmtop``` and ```.inpcrd```
     """
     #print(argumentList)
-    options = "h:n:m:s:o:c:b:g:u:rq:e:d:a:t:l:p:v:D:"
-    long_options = ["help", "solutename", "main", "solvent", "output", "charge", "cubesize", "chargemethod", "spinmultiplicity", "srunuse","qmprogram","qmexe", "qmdir", "amberhome", "closeness","solventoff","solventfrcmod","validation", "runningdirectory"]
+    options = "h:m:n:s:o:c:b:g:u:r:q:e:d:a:t:l:p:v:D:"
+    long_options = ["help", "main","solutename", "solvent", "output", "charge", "cubesize", "chargemethod", "spinmultiplicity", "srunuse","qmprogram","qmexe", "qmdir", "amberhome", "closeness","solventoff","solventfrcmod", "validation", "runningdirectory"]
```
Comment (ffangliu) [inline autosolvate/autosolvate.py:657]: Why did you change `r` to `r:`? Is the option `srunuse` now taking in an argument? The column symbol indicates whether it is something that requires an argument following that option.

File: autosolvate/autosolvate.py [hunk 3]
```diff
@@ -696,23 +694,31 @@ def startboxgen(argumentList):
...
+                slu_netcharge = info[2]
+                solS=Solute(info[0], info[1], info[2], info[3])
+                cube_size = solS.get_box_length()
+                slu_spinmult = solS.get_spin_multiplicity()
+                charge_method = solS.get_methods()[0]
+            else:
+                solS = Solute("", "", slu_netcharge, solutexyz)
+                mol = next(pybel.readfile("xyz", solutexyz))
+                total_electrons = sum(atom.atomicnum for atom in mol.atoms)
+                slu_spinmult = (total_electrons - slu_netcharge) % 2 + 1
```
Comment (ffangliu) [inline autosolvate/autosolvate.py:719]: Does this modification overwrite the slu_spinmult provided by the user?

**Review by ffangliu — APPROVED** [review]

**Pull request merged**